File: backend/internal/domain/entity/inventory.go
package entity

import "time"

type Inventory struct {
	ID            int64
	ItemID        string
	ItemName      string
	TotalStock    int
	ReservedStock int
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

func (i Inventory) AvailableStock() int {
	return i.TotalStock - i.ReservedStock
}

File: frontend/src/types/inventory.ts
export type InventoryItem = {
  item_id: string
  item_name: string
  total_stock: number
  reserved_stock: number
  available_stock: number
}

export type InventoryListResponse = {
  items: InventoryItem[]
}

export type Reservation = {
  reservation_id: string
  item_id: string
  quantity: number
  expires_at: string
}

export type ConfirmResponse = {
  reservation_id: string
  confirmed_at: string
}

export type Notice = {
  type: 'success' | 'error' | 'info'
  message: string
}

export type ReserveInventoryPayload = {
  user_id: string
  item_id: string
  quantity: number
}
